gordo_components/watchman/endpoints_status.py

```diff
@@ -0,0 +1,291 @@
+            description of intended event structure. Needs to have at least a key
+            `object`and a key `type`. 'object' should be of type
+            `kubernetes.client.models.v1_service.V1Service`
+
+        Returns
+        -------
+
+        """
+        event_obj: kubernetes.client.models.v1_service.V1Service = event["object"]
+        model_name = event_obj.metadata.labels.get(
+            "applications.gordo.equinor.com/model-name", None
+        )
+        event_type = event["type"]
+        logger.info(f"Got K8s event for model {model_name} of type {event_type}")
+
+        if event_type == "DELETED":
+            job_name = job_name_for_model_update(model_name)
+            if model_name and self.scheduler.get_job(job_name):
+                logger.info(f"Removing job {job_name}")
+                if model_name in self.model_metadata:
+                    self.model_metadata[model_name] = self.model_metadata[
+                        model_name
+                    ]._replace(
+                        healthy=False, last_seen=datetime.now(pytz.utc).isoformat()
+                    )
+                self.scheduler.remove_job(job_id=job_name)
+        else:
+            if model_name:
+                self._schedule_update_for_model(model_name)
+            else:
+                logger.warning(
+                    "Got updated model-server service notification, but found "
+                    "no model-name"
+                )
+
+    def _schedule_update_for_model(self, model_name, seconds=5):
+        """Schedules a regular model-update job for the model, every `seconds`
+        seconds. If the job exists it will be updated with the new schedule,
+        and if it does not exist it will be added with next_run_time = now
+        and interval = `seconds`.
+
+        """
+        job_name = job_name_for_model_update(model_name)
+        if self.scheduler.get_job(job_name):
+            logger.info(
+                f"Updating scheduled job {job_name} to run every {seconds} seconds"
+            )
+            self.scheduler.reschedule_job(
+                job_id=job_name, trigger="interval", seconds=seconds
+            )
+        else:
+            logger.info(
+                f"Adding scheduled job {job_name} to run every {seconds} seconds"
+            )
+            self.scheduler.add_job(
+                func=self.update_model_metadata,
+                trigger="interval",
+                seconds=seconds,
+                coalesce=True,
+                max_instances=1,
+                id=job_name,
+                next_run_time=datetime.now(),
+                kwargs={"model_name": model_name},
+            )
+
+    def update_model_metadata(self, model_name: str):
+        """
+        Updates model metadata for a model, and if it succeeds then reschedule the
+        update job to happen every `self.update_model_interval` second.
```

Comment:
Well, it will run forever, but just not at 2 seconds :-p If something crashes, whats best, that it will rerun the update job forever, or that it will never do it? Cause I see two methods:
1. Schedule it at an interval, and then change it on success, or
2. Not schedule it at an interval, but reschedule on failure. 

I have gone for the former option, so if something goes bad we are "guaranteed" to retry. But I am definitely open for changing the timings for example, or even the method.

Reply:
I'm not much for the implicit dependency between two methods/funcs to manage the state of rescheduling the task.

What about option 2 with [events](https://apscheduler.readthedocs.io/en/latest/userguide.html#scheduler-events)? Then we can explicitly log an exception occurs and reschedule? If you think it's more trouble than it's worth, I'm cool with it.

Reply:
It is definitely a bit trouble, and I think it is lame doing more on this PR I am sick of. But I agree that it is a bit ugly that `update_model_metadata` has the side-effect of updating the scheduler. I will look into the events and see if I can make that nice

Reply:
I changed my mind again